lib/templates/completion.zsh

```diff
@@ -13,6 +14,14 @@ if type compdef &>/dev/null; then
       _describe 'values' reply
     fi
   }
-  compdef _{pkgname}_completion {pkgname}
```

Comment:
So I did a bit of research and found that `#compdef {pkgname}` replaces this. The use-cases I listed above would be affected.

Although, it seems that `#compdef {pkgname}` is strictly necessary for lazily loading completion file.

Therefore, I request that:
1. If lazy loading is strictly incompatible with the use-cases I listed above, please create new entry for `zsh-lazy` with new template file. The relevant files are `constants.js` and `installer.js`.
2. If there is a way to be compatible with all use-cases in a single file, please make the necessary modifications for them.

Reply:
That's what the `if` is for, for supporting both in the same file. When the file is `source`ed it will call `compdef` (With the `#compdef` line being just a comment), while when called by the completion system it will run the completion function.

I haven't tested appending to `.zshrc` though, which might require adjusting the condition.

Reply:
> I haven't tested appending to `.zshrc` though, which might require adjusting the condition.

Please do. And tell me the result.

Reply:
Seems to work fine in `.zshrc` as well. In `.zshrc` `$zsh_eval_context` is file, which will make the snippet call `compdef` as needed.